pkg/driver/controller.go

```diff
@@ -286,19 +286,21 @@ func (d *controllerService) ControllerPublishVolume(ctx context.Context, req *cs
 	if !d.cloud.IsExistInstance(ctx, nodeID) {
 		return nil, status.Errorf(codes.NotFound, "Instance %q not found", nodeID)
 	}
-
-	if _, err := d.cloud.GetDiskByID(ctx, volumeID); err != nil {
+	disk, err := d.cloud.GetDiskByID(ctx, volumeID)
+	if err != nil {
 		if err == cloud.ErrNotFound {
 			return nil, status.Error(codes.NotFound, "Volume not found")
 		}
 		return nil, status.Errorf(codes.Internal, "Could not get volume with ID %q: %v", volumeID, err)
 	}
 
+	// If given volumeId already assigned to given node, will directly return current device path
 	devicePath, err := d.cloud.AttachDisk(ctx, volumeID, nodeID)
 	if err != nil {
-		if err == cloud.ErrAlreadyExists {
-			return nil, status.Error(codes.AlreadyExists, err.Error())
+		if err == cloud.ErrVolumeInUse {
+			return nil, status.Error(codes.FailedPrecondition, strings.Join(disk.Attachments, ","))
```

Comment:
I don't think this is what we want. Looking at the API documentation[[1]][[2]], this list will have information about previous attachments as well, but the status will be "detached". So I think what we want to do is filter for "attached". 

[1]: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Volume.html
[2]: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_VolumeAttachment.html

Reply:
Yeah, this is a good point, will update